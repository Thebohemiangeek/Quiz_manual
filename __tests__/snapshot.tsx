/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import LandingPage from "app/landing-page/page";
import { Suspense } from "react";

const mockQuizData = {
  questions: [
    {
      question: "What is your favorite color?",
      type: "single-choice",
      options: [
        { display: "Red", value: "red", isRejection: false },
        { display: "Blue", value: "blue", isRejection: false },
        { display: "Green", value: "green", isRejection: false },
      ],
    },
    {
      question: "What is your favorite animal?",
      type: "single-choice",
      options: [
        { display: "Cat", value: "cat", isRejection: false },
        { display: "Dog", value: "dog", isRejection: false },
        { display: "Bird", value: "bird", isRejection: false },
      ],
    },
  ],
};

interface LandingPageProps {
  quizData: typeof mockQuizData;
}

it("renders LandingPage unchanged", () => {
  const { container } = render(
    <Suspense fallback={<div>Loading...</div>}>
      <LandingPage quizData={mockQuizData} />
    </Suspense>
  );
  expect(container).toMatchSnapshot();
});
