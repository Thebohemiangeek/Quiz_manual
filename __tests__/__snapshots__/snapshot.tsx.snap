// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LandingPage unchanged 1`] = `
<div>
  <div>
    <section
      class="hero"
    >
      <img
        alt="Manual.co"
        class="hero__logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="40"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <div
        class="hero__caption"
      >
        <h1
          class="hero__title"
        >
          Be good 
          <br />
          to yourself
        </h1>
        <p
          class="hero__text"
        >
          We're working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
        </p>
        <button
          class="button   "
        >
          Take the Quiz
        </button>
      </div>
    </section>
    <div
      class="section__wrapper"
    >
      <h1
        class="section_title"
      >
        What we can help with
      </h1>
      <section
        class="section "
        data-testid="section-content"
      >
        <img
          alt="Image depicting hair loss"
          class="section_image"
          data-nimg="1"
          decoding="async"
          height="445"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=750&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=750&q=75 2x"
          style="color: transparent;"
          width="370"
        />
        <div
          class="section_textContent "
          data-section="01"
        >
          <span
            class="section_topic"
          >
            HAIR LOSS
          </span>
          <h3
            class="section_title"
          >
            Hair loss needn’t be irreversible. We can help!
          </h3>
          <p
            class="section_text"
          >
            We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
          </p>
        </div>
      </section>
      <section
        class="section section--reversed"
        data-testid="section-content"
      >
        <img
          alt="Image depicting erectile dysfunction"
          class="section_image"
          data-nimg="1"
          decoding="async"
          height="445"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=750&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=750&q=75 2x"
          style="color: transparent;"
          width="370"
        />
        <div
          class="section_textContent section_textContent--reversed"
          data-section="02"
        >
          <span
            class="section_topic"
          >
            ERECTILE DYSFUNCTION
          </span>
          <h3
            class="section_title"
          >
            Erections can be a tricky thing. But no need to feel down!
          </h3>
          <p
            class="section_text"
          >
            We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
          </p>
        </div>
      </section>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer_content"
      >
        <img
          alt="Manual.co"
          class="footer_logo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="footer_columns"
        >
          <ul
            class="footer_column"
          >
            <li
              class="footer_column__title"
            >
              Product
            </li>
            <li>
              Popular
            </li>
            <li>
              Trending
            </li>
            <li>
              Guide
            </li>
            <li>
              Products
            </li>
          </ul>
          <ul
            class="footer_column"
          >
            <li
              class="footer_column__title"
            >
              Company
            </li>
            <li>
              Press
            </li>
            <li>
              Mission
            </li>
            <li>
              Strategy
            </li>
            <li>
              About
            </li>
          </ul>
          <ul
            class="footer_column"
          >
            <li
              class="footer_column__title"
            >
              Info
            </li>
            <li>
              Support
            </li>
            <li>
              Customer Service
            </li>
            <li>
              Get started
            </li>
          </ul>
          <div>
            <span
              class="footer_socialMedia__title"
            >
              Follow us
            </span>
            <div
              class="footer_socialMedia__list"
            >
              <svg
                data-testid="fa-google"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 488 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                />
              </svg>
              <svg
                data-testid="fa-facebook-f"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
              <svg
                data-testid="fa-twitter"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="footer_separator"
      />
      <section
        class="copyright"
      >
        <span
          class="copyright__text"
          data-testid="copyright__text"
        >
          © 
          2024
           Manual. All rights reserved
        </span>
      </section>
    </footer>
  </div>
</div>
`;
